src/selectors/per-thread/markers.js

```diff
@@ -399,41 +392,44 @@ export function getMarkerSelectorsPerThread(
   );
 
   /**
-   * This returns only FileIO markers for the header.
-   * Also excludes FileIO markers that belong to other threads.
+   * This returns markers for the FileIO timeline. The Marker Schema is obeyed, but
+   * there is special handling to ensure that the FileIO markers are displayed
+   * only for that thread.
    */
-  const getFileIoMarkerIndexesForHeader: Selector<
+  const getTimelineFileIoMarkerIndexes: Selector<
     MarkerIndex[]
   > = createSelector(
     getMarkerGetter,
     getCommittedRangeAndTabFilteredMarkerIndexes,
-    filterMarkerIndexesCreator(MarkerData.isOnThreadFileIoMarker)
+    ProfileSelectors.getMarkerSchema,
+    () => 'timeline-fileio',
+    // Custom filtering in addition to the schema logic:
+    () => MarkerData.isOnThreadFileIoMarker,
+    MarkerData.filterMarkerByDisplayLocation
```

Comment:
For that one I feel like we could filter only using `MarkerData.isOnThreadFileIoMarker`, and that `filterMarkerByDisplayLocation` is only a distraction. Unless I miss something and there could be other types of marker in `timeline-fileio`?

BTW I noticed that FileIO markers have nothing as their `display` property, that's probably a mistake because this means they're displayed nowhere? (unless the chart and table aren't yet hooked to this :-) )

Reply:
If I take it out of the machinery, then it means you CAN'T use marker schema to target markers from Gecko to this track. I think it would be better to treat each track identically, even if the schema part is not being used. This makes this code more generic.

Reply:
> BTW I noticed that FileIO markers have nothing as their display property

Ah, yes it does look like a mistake.